java-client/src/main/java/org/opensearch/client/json/jackson/JacksonJsonProvider.java

```diff
@@ -67,7 +67,7 @@
     private JsonFactory jsonFactory = new JsonFactory();
 
     public JacksonJsonProvider(JsonFactory jsonFactory) {
-        this.jsonFactory = new JsonFactory();
+        this.jsonFactory = jsonFactory;
```

Comment:
Seems to be a mistake, `jsonFactory` was never taken into account

Reply:
FAR: should line 67 be: private JsonFactory jsonFactory;

Reply:
And even `final`, thank you.